spec/bundler/ruby_version_spec.rb

```diff
@@ -1,80 +1,390 @@
-    req = requirement("2.0.0", "645", "ruby", "2.0.1")
-    ver = requirement("2.0.0", "643", "ruby", "2.0.0")
+      context "the versions, pathlevels, engines, and engine_versions match" do
+        it "should return true" do
+          expect(subject).to eq(other_ruby_version)
+        end
+      end
 
-    expect(req.diff(ver)).to eq([:engine_version, "2.0.1", "2.0.0"])
-  end
+      context "the versions do not match" do
+        let(:other_version) { "1.21.6" }
+
+        it_behaves_like "two ruby versions are not equal"
+      end
+
+      context "the patchlevels do not match" do
+        let(:other_patchlevel) { "21" }
+
+        it_behaves_like "two ruby versions are not equal"
+      end
+
+      context "the engines do not match" do
+        let(:other_engine) { "ruby" }
+
+        it_behaves_like "two ruby versions are not equal"
+      end
+
+      context "the engine versions do not match" do
+        let(:other_engine_version) { "1.11.2" }
+
+        it_behaves_like "two ruby versions are not equal"
+      end
+    end
 
-  it "detects patchlevel discrepancies last" do
-    req = requirement("2.0.0", "645", "ruby", "2.0.1")
-    ver = requirement("2.0.0", "643", "ruby", "2.0.1")
+    describe "#host" do
+      before do
+        allow(RbConfig::CONFIG).to receive(:[]).with("host_cpu").and_return("x86_64")
+        allow(RbConfig::CONFIG).to receive(:[]).with("host_vendor").and_return("apple")
+        allow(RbConfig::CONFIG).to receive(:[]).with("host_os").and_return("darwin14.5.0")
+      end
 
-    expect(req.diff(ver)).to eq([:patchlevel, "645", "643"])
+      it "should return an info string with the host cpu, vendor, and os" do
+        expect(subject.host).to eq("x86_64-apple-darwin14.5.0")
+      end
+
+      it "memoizes the info string with the host cpu, vendor, and os" do
+        expect(RbConfig::CONFIG).to receive(:[]).with("host_cpu").once.and_call_original
+        expect(RbConfig::CONFIG).to receive(:[]).with("host_vendor").once.and_call_original
+        expect(RbConfig::CONFIG).to receive(:[]).with("host_os").once.and_call_original
+        2.times { ruby_version.host }
+      end
+    end
+
+    describe "#gem_version" do
+      let(:gem_version_obj) { Gem::Version.new(version) }
+
+      it "should return a Gem::Version instance with the correct version" do
+        expect(ruby_version.gem_version).to eq(gem_version_obj)
+        expect(ruby_version.gem_version.version).to eq("2.0.0")
+      end
+    end
   end
 
-  it "successfully matches gem requirements" do
-    req = requirement(">= 2.0.0", "< 643", "ruby", "~> 2.0.1")
-    ver = version("2.0.0", "642", "ruby", "2.0.5")
+  describe Bundler::SystemRubyVersion do
+    subject { Bundler::SystemRubyVersion.new(version, patchlevel, engine, engine_version) }
+
+    let(:system_ruby_version) { subject }
+
+    describe "#initialize" do
+      # Suppress "warning: instance variable @variable not initialized" warnings
```

Comment:
ew. TBH, looking at that code, I'd kinda like to refactor it to just call through to the super initializer? It could even just be a factory method and we could probably get rid of the whole subclass